system/View/Interfaces/ViewDecorator.php

```diff
@@ -0,0 +1,30 @@
+<?php
+
+/**
+ * This file is part of CodeIgniter 4 framework.
+ *
+ * (c) CodeIgniter Foundation <admin@codeigniter.com>
+ *
+ * For the full copyright and license information, please view
+ * the LICENSE file that was distributed with this source code.
+ */
+
+namespace CodeIgniter\View\Interfaces;
+
+/**
+ * View Decorators are simple classes that are given the
+ * chance to modify the output from the view() calls
+ * prior to it being cached.
+ */
+interface ViewDecorator
+{
+    /**
+     * Takes $html and has a chance to alter it.
+     * MUST return the modified HTML.
+     *
+     * @param string $html
+     *
+     * @return string
+     */
+    public function decorate(string $html): string;
```

Comment:
Since the execution of this method always mounts to `(new $class())->decorate($html)` is there any reason not to make this static and just call it without ever instantiating the class?
```suggestion
    public static function decorate(string $html): string;
```

Reply:
Done